### PR TITLE
EVA-1173 Create job and step in accessioning pipeline

### PR DESCRIPTION
**Pull request by andresfsilva**

### CODE REVIEW

File: eva-accession-pipeline/src/test/java/uk/ac/ebi/eva/accession/pipeline/io/AccessionWriterTest.java [hunk 1]
```diff
@@ -40,8 +38,8 @@
 import static org.junit.Assert.assertTrue;
 
 @RunWith(SpringRunner.class)
-@DataJpaTest
-@ContextConfiguration(classes = {SubmittedVariantAccessioningConfiguration.class, InputParametersConfiguration.class})
+@AutoConfigureDataJpa
```
Comment (cyenyxe) [inline eva-accession-pipeline/src/test/java/uk/ac/ebi/eva/accession/pipeline/io/AccessionWriterTest.java]: For other reviewers, I was wondering about this change and then noticed that [@DataJpaTest](https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/test/autoconfigure/orm/jpa/DataJpaTest.html) involves what `@AutoConfigureDataJpa` does and quite a lot more.

File: eva-accession-pipeline/src/test/resources/accession-pipeline-test.properties [hunk 2]
```diff
@@ -3,3 +3,13 @@ accessioning.variant.blockSize=1000
 accessioning.variant.categoryId=test-ss
 
 accessioning.monotonic.init.test-ss=10000000000
+
+parameters.assemblyAccession=assembly
+parameters.taxonomyAccession=1111
+parameters.projectAccession=project
+parameters.chunkSize=100
```
Comment (cyenyxe) [inline eva-accession-pipeline/src/test/resources/accession-pipeline-test.properties]: It would make sense to merge `chunkSize` and `blockSize` in a single parameter. If we don't do that in this ticket a comment should be added to the application runner JIRA ticket.

File: eva-accession-pipeline/src/test/java/uk/ac/ebi/eva/accession/pipeline/test/BatchTestConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    @PostConstruct
+    protected void initialize() {
+        if (this.properties.getInitializer().isEnabled()) {
+            String platform = "h2";
+            ResourceDatabasePopulator populator = new ResourceDatabasePopulator();
+            String schemaLocation = this.properties.getSchema();
+            schemaLocation = schemaLocation.replace("@@platform@@", platform);
+            populator.addScript(this.resourceLoader.getResource(schemaLocation));
+            populator.setContinueOnError(true);
```
Comment (cyenyxe) [inline eva-accession-pipeline/src/test/java/uk/ac/ebi/eva/accession/pipeline/test/BatchTestConfiguration.java]: It would be safer if the default value (false) for this flag was used, because according to the documentation:

> Flag to indicate that all failures in SQL should be logged but not cause a failure.

File: eva-accession-pipeline/src/test/java/uk/ac/ebi/eva/accession/pipeline/test/BatchTestConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    private PlatformTransactionManager platformTransactionManager;
+
+    @Bean
+    public JobLauncherTestUtils jobLauncherTestUtils() {
+        return new JobLauncherTestUtils();
+    }
+
+    @PostConstruct
+    protected void initialize() {
+        if (this.properties.getInitializer().isEnabled()) {
```
Comment (cyenyxe) [inline eva-accession-pipeline/src/test/java/uk/ac/ebi/eva/accession/pipeline/test/BatchTestConfiguration.java]: Is this really necessary? I can't find anything similar in any of our other pipeline projects.

File: eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/configuration/BeanNames.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+ */
+package uk.ac.ebi.eva.accession.pipeline.configuration;
+
+public class BeanNames {
+
+    public static final String ACCESSION_READER = "VARIANT_READER";
+
+    public static final String ACCESSION_PROCESSOR = "ACCESSION_PROCESSOR";
+
+    public static final String ACCESSION_WRITER = "ACCESSION_WRITER";
```
Comment (jmmut) [inline eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/configuration/BeanNames.java]: this should be used in AccessionWriterConfiguration

File: eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/configuration/BeanNames.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.eva.accession.pipeline.configuration;
+
+public class BeanNames {
+
+    public static final String ACCESSION_READER = "VARIANT_READER";
+
+    public static final String ACCESSION_PROCESSOR = "ACCESSION_PROCESSOR";
```
Comment (jmmut) [inline eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/configuration/BeanNames.java]: Isn't it better VARIANT_PROCESSOR? it doesn't process accessions

File: eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/configuration/BeanNames.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.eva.accession.pipeline.configuration;
+
+public class BeanNames {
+
+    public static final String ACCESSION_READER = "VARIANT_READER";
```
Comment (jmmut) [inline eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/configuration/BeanNames.java]: VARIANT_READER fits better

File: eva-accession-pipeline/src/test/java/uk/ac/ebi/eva/accession/pipeline/test/BatchTestConfiguration.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    @Autowired
+    private PlatformTransactionManager platformTransactionManager;
+
+    @Bean
+    public JobLauncherTestUtils jobLauncherTestUtils() {
+        return new JobLauncherTestUtils();
+    }
+
+    @PostConstruct
+    protected void initialize() {
```
Comment (cyenyxe) [inline eva-accession-pipeline/src/test/java/uk/ac/ebi/eva/accession/pipeline/test/BatchTestConfiguration.java]: Add a method comment explaining why is this necessary, because normally it shouldn't be.
